apps/nextjs-website/react-components/components/Footer/helpers/LangSwitch.tsx

```diff
@@ -0,0 +1,82 @@
+import { useRef, useState } from 'react';
+import { Menu, MenuItem, Typography, Box, Stack, Button } from '@mui/material';
+import {
+  KeyboardArrowDownRounded,
+  KeyboardArrowUpRounded,
+} from '@mui/icons-material';
+import { LangSwitchProps } from '../../../types/Footer/Footer.types';
+
+export function LangSwitch({
+  languages,
+  onLanguageChanged,
+  activeLanguage,
+}: LangSwitchProps) {
+  const [menuOpen, setMenuOpen] = useState(false);
+
+  const toggleMenu = () => {
+    setMenuOpen((open) => !open);
+  };
+
+  const anchorEl = useRef(null);
+
+  return (
+    <Stack display='flex' flexDirection='column' aria-label='cambia la lingua'>
+      <Button
+        sx={{
+          fontWeight: '600',
+          letterSpacing: 0.3,
+          color: 'text.primary',
+          height: 'auto',
+          display: 'flex',
+          padding: 0,
+          marginBottom: 1,
+          '&:hover': {
+            backgroundColor: 'transparent',
+          },
+        }}
+        variant='text'
+        size='small'
+        disableRipple
+        aria-label='lingua'
+        aria-haspopup='true'
+        aria-expanded={menuOpen}
+        onClick={toggleMenu}
+      >
+        <Box ref={anchorEl}>
+          <Typography color='inherit' variant='subtitle2'>
+            {activeLanguage.value}
+          </Typography>
+        </Box>
+
+        {menuOpen ? (
+          <KeyboardArrowDownRounded fontSize='small' />
+        ) : (
+          <KeyboardArrowUpRounded fontSize='small' />
+        )}
+      </Button>
+      {!!languages?.length && anchorEl && (
```

Comment:
```suggestion
      {languages.length > 0 && anchorEl && (
```
`languages` is not optional